crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -383,7 +383,8 @@ impl IRGenerator {
                 Value::Single(v_id)
             }
             Type::String(len) => {
-                let obj_type = ObjectType::Unsigned(8);
+                // TODO: document why this is 8
```

Comment:
Minor nit here, note that if UTF8 is used then a character can be `1-4 bytes`, so we should be clear about the encoding and the layout

Reply:
A character can be 1-N bytes really as rust's `char` type is only for Unicode scalar values. I think it is sufficient to say we encode strings as a packed array of bytes encoded in utf-8.

Reply:
Sounds good